project_definition/models/document_page.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class DocumentPage(models.Model):
+    _inherit = 'document.page'
+
+    # a relation to partner_id is given by our dependency on
+    # document_page_partner_id if there is no partner_id and no project_id
+    # specified the term is global
+
+    type = fields.Selection(selection_add=[('term', 'Term')])
+
+    # stored compute field used for making the letter grouping
+    # need to be stored in order to use grouping
+    first_letter = fields.Char(compute='_get_first_letter', store=True)
```

Comment:
that's fine for now, but in the future consider other more elegant options:

- generate filters A-Z + rest in the search view, either manually or in `fields_view_get`
- use a custom group operator that does this
- have a module that adds A-Z + rest as buttons to the search view which will do a name_search('$letter%', operator='=ilike')
- have a module that overrides [`_read_group_process_groupby`](https://github.com/OCA/OCB/blob/8.0/openerp/models.py#L1945) to support some syntax like `$char_field:first`